src/test/java/com/fittracker/fittracker/controller/BaseIntegrationTest.java

```diff
@@ -102,6 +103,19 @@ protected <T> T makeRequestWithBody(HttpMethod httpMethod, Object requestBody, H
         return mapper.readValue(responseString, responseClass);
     }
 
+    protected <T> T makeRequestWithBodyAndPathVariable(HttpMethod httpMethod, Object requestBody,
```

Comment:
On second thought, what happens if we use `MockMvc` as we would like to make a request with body, but leave the body empty (passing null value or leaving body as vararg at the end of method signature). If it will behave as request without body, we can think of collapsing `makeRequest`, `makeRequestWithBody (both)` and `makeRequestWithBodyAndPathVariable` into one function. 

Let's have a chat about it at the nearest opportunity.